cmd/argocd-server/commands/root.go

```diff
@@ -98,6 +100,7 @@ func NewCommand() *cobra.Command {
 	command.Flags().BoolVar(&insecure, "insecure", false, "Run server without TLS")
 	command.Flags().StringVar(&staticAssetsDir, "staticassets", "", "Static assets directory path")
 	command.Flags().StringVar(&baseHRef, "basehref", "/", "Value for base href in index.html. Used if Argo CD is running behind reverse proxy under subpath different from /")
+	command.Flags().StringVar(&rootPath, "rootpath", "", "Used if Argo CD is running behind reverse proxy under subpath different from /")
```

Comment:
The `--rootpath` flag is supposed to change both server-side and client-side behavior. So if it is specified we need to add root path on server-side and change UI `basehref` as well. 

I think we should just check if `rootPath` if not empty and change `baseHRef` to the same value. If for some reason user has specified  conflicting`baseHRef` value then I suggest to print warning and override it with `rootPath`